src/services/validate-event.ts

```diff
@@ -3,37 +3,31 @@ import logger from '../commons/logger';
 import { logAndPublishMetric } from '../commons/metrics';
 import { MetricNames } from '../data-types/constants';
 import { ValidationError } from '../data-types/errors';
+import { compileSchema } from '../commons/compile-schema';
+import { TxMAIngress } from '../data-types/schemas';
+
+const validateInterventionDataInput = compileSchema(TxMAIngress);
 
 /**
  * A function to check the event has the necessary fields to continue with the processing.
  *
  * @param interventionRequest - the TxMA request
  */
 export function validateEvent(interventionRequest: TxMAEvent): void {
-  if (
-    !interventionRequest.timestamp ||
-    Number.isNaN(interventionRequest.timestamp) ||
-    !interventionRequest.user?.user_id ||
-    !interventionRequest.event_name
-  ) {
-    logger.debug('event has failed initial validation');
+  if (!validateInterventionDataInput({ event: interventionRequest })) {
+    logger.debug('event has failed schema validation');
```

Comment:
Can we change to capital letter and full stop 'Event has failed the schema validation.'?